tutorials/images/tracing/EditSequenceDiagramModelHighLevel.puml

```diff
@@ -0,0 +1,62 @@
+@startuml
+!include ../style.puml
+skinparam ArrowFontStyle plain
+
+box Logic LOGIC_COLOR_T1
+participant "EditCommand" as EditCommand LOGIC_COLOR
+participant "CommandResult" as CommandResult LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant "FilteredList<Person>" as ObservableList MODEL_COLOR
+participant "ModelManager" as Model MODEL_COLOR
+end box
+
+--> EditCommand
+activate EditCommand
+
+
+
+EditCommand -> Model : getFilteredPersonList()
+activate Model
+Model --> EditCommand
+deactivate Model
+
+
+EditCommand --> ObservableList : retrieve Person
+activate ObservableList
+ObservableList --> EditCommand : Person
+deactivate ObservableList
+
+EditCommand --> EditCommand : createEditedPerson
+activate EditCommand
+EditCommand --> EditCommand
+deactivate EditCommand
+
+EditCommand --> Model : setPerson
+activate Model
+Model --> EditCommand
+deactivate Model
+
+EditCommand --> Model : updateFilteredPersonList
+activate Model
+Model --> ObservableList : setPredicate
+activate ObservableList
+ObservableList --> Model
+deactivate ObservableList
+Model --> EditCommand
+deactivate Model
+
+
+
+
+create CommandResult
+EditCommand --> CommandResult
+activate CommandResult
+CommandResult --> EditCommand
+deactivate CommandResult
+
+[<-- EditCommand
+deactivate EditCommand
+
```

Comment:
Suggested: 
![image](https://github.com/user-attachments/assets/905ec2f7-2a48-4c41-bc77-1a3a89d43281)


Changes:
1. Moved FilteredPersonList out of Model. Although the class is in the Model, and the object is kept as a variable inside ModelManager, it simply acts as a data transfer object between the two components. So, better to show outside of both components. This also solves the problem of more than one component showing low-level details (as the Model now does not show low level details).
1. Added object names, semicolon before class name
1. Used solid arrow for method calls
1. Show parameters and return values where useful
1. Removed `setPredicate(...)` as that's an internal detail of the Model.

You can tweak further, as needed.

```puml
@startuml
!include ../style.puml
skinparam ArrowFontStyle plain

box Logic LOGIC_COLOR_T1
participant ":EditCommand" as EditCommand LOGIC_COLOR
participant "r:CommandResult" as CommandResult LOGIC_COLOR
end box

participant "filteredPersons:FilteredList<Person>" as ObservableList MODEL_COLOR
box Model MODEL_COLOR_T1
participant "model:ModelManager" as Model MODEL_COLOR
end box

-> EditCommand : execute(model)
activate EditCommand



EditCommand -> Model : getFilteredPersonList()
activate Model
Model --> EditCommand : filteredPersons
deactivate Model


EditCommand -> ObservableList : get(index)
activate ObservableList
ObservableList --> EditCommand : personToEdit
deactivate ObservableList

EditCommand -> EditCommand : createEditedPerson(personToEdit, editPersonDescriptor)
activate EditCommand
EditCommand --> EditCommand: editedPerson
deactivate EditCommand

EditCommand -> EditCommand : check for duplicates
activate EditCommand
deactivate EditCommand

EditCommand -> Model : setPerson(personToEdit, editedPerson)
activate Model
Model --> EditCommand
deactivate Model

EditCommand -> Model : updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS)
activate Model
deactivate ObservableList
Model --> EditCommand
deactivate Model


create CommandResult
EditCommand -> CommandResult
activate CommandResult
CommandResult --> EditCommand
deactivate CommandResult

[<-- EditCommand: r
deactivate EditCommand

@enduml

```

Reply:
Thanks Prof, will update existing puml with this and work on it as needed.

Reply:
I think I will omit the arguments as it can referred in the code and will not come out too overwhelming